apis/python/tests/test_collection.py

```diff
@@ -249,3 +250,38 @@ def test_exceptions_on_not_created(tmp_path):
         list(sc)
     with pytest.raises(DoesNotExistError):
         len(sc)
+
+
+def test_collection_timestamped_reads(tmp_path):
+    """
+    When we have a collection open at a given timestamp, accessed elements inherit that timestamp
+    for reading.
+    """
+    basedir = tmp_path.as_uri()
+    collection = soma.Collection(basedir)
+    collection.create()
+
+    dataframe = soma.DataFrame(os.path.join(basedir, "sdf"), parent=collection)
+    create_and_populate_dataframe(dataframe)
+    sparse_nd_array = soma.SparseNDArray(
+        os.path.join(basedir, "snda"), parent=collection
+    )
+    create_and_populate_sparse_nd_array(sparse_nd_array)
+
+    collection.set("sdf", dataframe)
+    collection.set("snda", sparse_nd_array)
+
+    time.sleep(0.01)
```

Comment:
I'd recommend against sleeping and instead use write + read timestamps to guarantee ordering.